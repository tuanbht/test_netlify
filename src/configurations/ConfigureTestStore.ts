import { configureStore, EnhancedStore } from '@reduxjs/toolkit';
import MockAdapter from 'axios-mock-adapter';
import axiosMiddleware from 'redux-axios-middleware';
import RootReducer from '../reducers';
import axiosClient from './ApiClient';

export const testStore = (initialState = {}): EnhancedStore =>
  configureStore({
    preloadedState: initialState,
    reducer: RootReducer,
    middleware: [axiosMiddleware(axiosClient)],
  });

export const mockAxios = new MockAdapter(axiosClient);
