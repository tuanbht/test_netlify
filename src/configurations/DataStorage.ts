import { ORDER_ID, PAYMENT_TOKEN } from '../constants/StoreKeys';

const setToken = (token: string): void => {
  localStorage.setItem(PAYMENT_TOKEN, token);
};

const setOrderId = (orderId: number): void => {
  localStorage.setItem(ORDER_ID, String(orderId));
};

const getToken = (): string | null => {
  return localStorage.getItem(PAYMENT_TOKEN);
};

const getOrderId = (): number => {
  return parseInt(localStorage.getItem(ORDER_ID) as string);
};

const clearAll = (): void => {
  localStorage.clear();
};

export default {
  setOrderId,
  setToken,
  getToken,
  getOrderId,
  clearAll,
};
