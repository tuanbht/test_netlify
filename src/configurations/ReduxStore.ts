import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { createBrowserHistory } from 'history';
import axiosMiddleware from 'redux-axios-middleware';
import axiosClient from './ApiClient';
import rootReducer from '../reducers';
import logger from 'redux-logger';

const history = createBrowserHistory();

const middleware = [
  axiosMiddleware(axiosClient),
  ...getDefaultMiddleware({
    serializableCheck: false,
  }),
  logger,
];

const store = configureStore({
  reducer: rootReducer,
  middleware: middleware,
});

export { history, store };
