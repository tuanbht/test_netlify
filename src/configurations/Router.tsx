import React from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';
import { NOT_FOUND_PATH, ORDER_CANCELLED_PATH, ORDER_EXPIRED_PATH, ORDER_PATH, ROOT_PATH } from 'constants/RouterPaths';
import { ORDER_CANCELLED_CONTENT, ORDER_EXPIRED_CONTENT } from 'constants/NotificationContent';
import { CustomerPayments, NotFound, NotificationPage, HomePage } from 'containers';

export const Router = (): React.ReactElement => {
  return (
    <Switch>
      <Route exact path={ROOT_PATH} component={HomePage} />
      <Route exact path={ORDER_CANCELLED_PATH} render={() => <NotificationPage content={ORDER_CANCELLED_CONTENT} />} />
      <Route exact path={ORDER_EXPIRED_PATH} render={() => <NotificationPage content={ORDER_EXPIRED_CONTENT} />} />
      <Route exact path={ORDER_PATH} component={CustomerPayments} />
      <Route exact path={NOT_FOUND_PATH} component={NotFound} />
      <Route render={() => <Redirect to={NOT_FOUND_PATH} />} />
    </Switch>
  );
};

export default Router;
