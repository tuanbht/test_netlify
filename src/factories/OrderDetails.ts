import { ORDER_STATUS, OrderDetails } from '../constants/CustomerPayments';
import faker from 'faker';

export const HardCodedOrderDetails = {
  status: 'order-status',
  orderNumber: 1,
  price: 1.0,
  storeName: 'store-name',
  storePhoneNumber: 'store-phone-number',
  markAsPaid: true,
  markAsPaidTime: 1601607445,
} as OrderDetails;

export const OrderDetailsEntity = {
  status: ORDER_STATUS.initialized,
  orderNumber: faker.random.number(),
  price: faker.random.number(),
  storeName: faker.company.companyName(),
  storePhoneNumber: faker.phone.phoneNumber(),
  markAsPaid: true,
  markAsPaidTime: new Date().getTime() / 1000,
} as OrderDetails;

export const buildOrderDetailsResponse = (props: {
  status?: string;
  hasStoreDetails?: boolean;
  hasCryptoETH?: boolean;
  hasCryptoBTC?: boolean;
  hasCryptoUSDT?: boolean;
  markAsPaid?: boolean;
}): any => ({
  data: {
    type: 'orders',
    attributes: {
      'external-order-id': 'NjKvLP1o',
      price: '67.77',
      'phone-number': '1-924-340-4717',
      name: 'Peggy Nikolaus',
      status: props.status,
      'marked-as-paid': props.markAsPaid,
      'marked-as-paid-at': props.markAsPaid && 1601607445,
    },
    relationships: {
      store: {
        data: {
          type: 'stores',
          id: '2',
        },
      },
      payments: {
        data: [
          {
            type: 'payments',
            id: '3',
          },
          {
            type: 'payments',
            id: '4',
          },
        ],
      },
    },
  },
  included: [
    props.hasStoreDetails && {
      id: '2',
      type: 'stores',
      links: {
        self: 'http://192.168.0.80:3000/v1/stores/2',
      },
      attributes: {
        name: 'Marilu Harvey MD',
        address: 'Suite 612 42874 Timmy Drive, Kerlukechester, IN 43320',
        'phone-number': '993.885.9672',
      },
      relationships: {
        merchant: {
          links: {
            self: 'http://192.168.0.80:3000/v1/stores/2/relationships/merchant',
            related: 'http://192.168.0.80:3000/v1/stores/2/merchant',
          },
        },
        orders: {
          links: {
            self: 'http://192.168.0.80:3000/v1/stores/2/relationships/orders',
            related: 'http://192.168.0.80:3000/v1/stores/2/orders',
          },
          data: [
            {
              type: 'orders',
              id: '2',
            },
          ],
        },
      },
    },
    props.hasCryptoETH && {
      id: '3',
      type: 'payments',
      attributes: {
        amount: '0.47646764203043',
        'destination-wallet': '0x3f5CE5FBFe3E9af3971dD833D26bA9b5C936f0bE',
        'payment-type': 'ETH',
      },
    },
    props.hasCryptoBTC && {
      id: '4',
      type: 'payments',
      attributes: {
        amount: '0.755828606587089',
        'destination-wallet': '1A1zP1eP5QGefi2DMPTfTL5SLmv7DivfNa',
        'payment-type': 'BTC',
      },
    },
    props.hasCryptoUSDT && {
      id: '5',
      type: 'payments',
      attributes: {
        amount: '0.755828606587089',
        'destination-wallet': '1A1zP1eP5QGefi2DMPTfTL5SLmv7DivfNa',
        'payment-type': 'USDT',
      },
    },
  ],
});
