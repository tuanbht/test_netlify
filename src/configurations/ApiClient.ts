import Axios from 'axios';
import { API_URL } from 'constants/ApiPaths';
import { NOT_FOUND_PATH } from 'constants/RouterPaths';
import { store } from './ReduxStore';

const axiosClient = Axios.create({
  baseURL: API_URL,
  headers: {
    Accept: 'application/vnd.api+json',
    'Content-Type': 'application/vnd.api+json',
  },
});

axiosClient.interceptors.request.use((config) => {
  const token = store.getState().Credential.token;
  if (token) config.headers['Payment-Token'] = token;
  else window.location.assign(NOT_FOUND_PATH);
  return config;
});

export default axiosClient;
