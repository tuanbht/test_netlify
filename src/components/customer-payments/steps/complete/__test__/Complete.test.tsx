import { mount } from 'enzyme';
import React from 'react';
import Complete from '../index';
import { Provider } from 'react-redux';
import { testStore } from '../../../../../configurations/ConfigureTestStore';
import { HardCodedOrderDetails } from '../../../../../factories/OrderDetails';

describe('Complete', () => {
  const store = testStore({
    OrderDetails: HardCodedOrderDetails,
  });

  it('renders template correctly', () => {
    expect(
      mount(
        <Provider store={store}>
          <Complete />
        </Provider>,
      ),
    ).toMatchSnapshot();
  });
});
