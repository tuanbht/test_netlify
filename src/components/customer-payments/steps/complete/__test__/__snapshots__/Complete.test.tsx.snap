// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Complete renders template correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Complete>
    <div>
      <div
        className="makeStyles-root-1"
      >
        <div
          className="makeStyles-orderLayout-2"
        >
          <div>
            <div
              className="makeStyles-merchantName-3"
            >
              store-name
            </div>
            <div
              className="makeStyles-orderName-4"
            >
              Order #
              1
            </div>
          </div>
          <div
            className="makeStyles-amount-5"
          >
            $
            1
          </div>
        </div>
        <div
          className="makeStyles-messageLayout-6"
        >
          <div
            className="makeStyles-titleMessage-7"
          >
            <ForwardRef(SvgIconChecked)
              className="makeStyles-iconChecked-8"
            >
              <svg
                className="makeStyles-iconChecked-8"
              >
                icon-checked.svg
              </svg>
            </ForwardRef(SvgIconChecked)>
            <span>
              Order #
              1
            </span>
          </div>
          <div
            className="makeStyles-subMessage1-9"
          >
            Your order has been confirmed and is out for delivery!
          </div>
          <div
            className="makeStyles-subMessage2-10"
          >
            If you have an issues, please contact 
            store-name
             at 
            store-phone-number
            .
          </div>
        </div>
      </div>
    </div>
  </Complete>
</Provider>
`;
